includes/controller/api.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * Creates a link or url to a given resource.
+ *
+ * @param $resource
+ */
+function api_link($resource)
+{
+    return page_link_to('api') . '&r=' . $resource;
+}
+
+function getApiShifts()
+{
+    $pageSize = isset($_REQUEST['per_page']) && preg_match('/\d+/', $_REQUEST['per_page']) ? (int)$_REQUEST['per_page'] : null;
+    $page = isset($_REQUEST['page']) && preg_match('/\d+/', $_REQUEST['page']) ? (int)$_REQUEST['page'] : null;
+    $limit = '';
+    if (null !== $page && null !== $pageSize) {
+        $offset = $page === 1 ? 0 : $page === 2 ? $pageSize : ($page - 1)*$pageSize;
```

Comment:
@ElectricMaxxx für ```$page = 0``` ändert mein Vorschlag doch nichts - den Fall sollte man aber in der Tat behandeln :)